api/src/accounts.rs

```diff
@@ -118,7 +120,19 @@ impl AccountsApi {
                 ledger_version.0,
                 start.0.map(StateKey::from),
                 limit.0,
+                false,
             )?;
+            if let Some(version) = ledger_version.0 {
+                if account.ledger_version < Version::from(version) {
+                    return Err(BasicErrorWith404::not_found_with_code_no_info(
+                        format!(
+                            "Request version {} larger than latest indexer version {}",
+                            version, account.ledger_version
+                        ),
+                        AptosErrorCode::VersionNotFound,
+                    ));
+                }
+            }
```

Comment:
shouldn't these have been checked already in the Account constructor?

Reply:
removed